### ISSUE TITLE
Bug: Empty strings are removed in JSONInput

### ISSUE THREAD

**Issue by znicholasbrown** [PrefectHQ/prefect-ui-library#1558]
When typing in the `JSONInput` component as part of schemas (or any other forms), validation seems to be removing empty strings as you type, converting them instead to empty objects.

**Comment by pleek91** [comment]
@znicholasbrown I'm not sure how to reproduce this. Do you mind adding a more detailed reproduction?

**Comment by znicholasbrown** [comment]
Yup sorry about that; any JSON input will work, I've used env vars on a new K8s work pool as an example:
https://github.com/PrefectHQ/prefect-ui-library/assets/27291717/d43b59ce-c275-4211-9b6b-8030941de3c4

As you'll see in that video, whenever I use top-level quotes, the input automatically removes them and unwraps their content. This behavior is only for top level quotes and makes sense given how we're evaluating those strings. That said, if we evaluate strings this way, bare text should also be valid but isn't.

**Comment by prefectcboyd** [comment]
Saw this reported and confirmed as well:
https://prefect-community.slack.com/archives/CM28LL405/p1688802308936069

Essentially the following is treated "incorrect" in the UI and quotes are removed, and I cannot submit the input / flow:
```
@flow
def main(input = None):
    print (input)


if __name__ == "__main__":
    main()
```

If I type `"test"` in the input block, the quotes are removed:
![image](https://github.com/PrefectHQ/prefect-ui-library/assets/105737359/e74b2b52-4310-4613-b83e-2e3bcdc1a80f)

If I type-hint `input: str = None` then this works fine.

"Test" is proper JSON, so I think it should be accepted, without requiring type-hinting?

**Comment by prefectcboyd** [comment]
Just as a follow-up - I can successfully workaround the issue by running this through the CLI, so it seems isolated to the field formatting in the UI.

E.g. with the above flow, I can run the following successfully via CLI:
`prefect deployment run 'main/input_test' --param input="test"`

**Comment by znicholasbrown** [comment]
Closed by #1563
